compiler/rustc_parse/src/parser/generics.rs

```diff
@@ -145,6 +145,19 @@ impl<'a> Parser<'a> {
                         } else {
                             (None, Vec::new())
                         };
+
+                        if this.eat(&token::Eq) {
+                            if this.check_lifetime() {
+                                let _lifetime = this.expect_lifetime();
+                                this.struct_span_err(
```

Comment:
Can you migrate this to a `derive(Diagnostic)`? https://rustc-dev-guide.rust-lang.org/diagnostics/diagnostic-structs.html